core/src/com/unciv/models/ruleset/Building.kt

```diff
@@ -495,6 +495,7 @@ class Building : RulesetStatsObject(), INonPerpetualConstruction {
             else -> {
                 if (filter == name) return true
                 if (filter == replaces) return true
+                if (hasTagUnique(filter)) return true
```

Comment:
No, this is in matchingFilter for a reason - tag uniques can now accept conditionals, so putting it here means the result will be cached, which is not what we want